### ISSUE TITLE
RFC: Reproducable training process

### ISSUE THREAD

**Issue by stweil** [mittagessen/kraken#302]
It looks like `ketos` currently shuffles the training data randomly, so I'd expect that a training process cannot be reproduced exactly.

I think that reproducable training is desirable.

It could be achieved by several alternatives:

- Add an option which disables the random shuffling. Users can still provide well defined shuffled lists of data.
- Use pseudo random shuffling with a seed which can be set by an option.

**Comment by mittagessen** [comment]
You can already set the numpy and torch seeds. `ketos -s 1234 ...`. To make it truly reproducible a lot more is involved though. Parallelized dataset parsing/loading changes the order (first changes the order with which items get added, second reinitializes the seed). We'd also need to set a bunch of other parameters (documented [here](https://pytorch.org/docs/stable/notes/randomness.html)) but some things such as non-deterministic LSTM layers are just an inherent feature of some CuDNN versions.

**Comment by amitdo** [comment]
https://pytorch.org/docs/stable/notes/randomness.html

https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#reproducibility

**Comment by mittagessen** [comment]
There has been deterministic training for a while now with the `-r` option to ketos.
